pandas/core/internals/managers.py

```diff
@@ -1821,6 +1847,18 @@ def _multi_blockify(tuples, dtype: Optional[Dtype] = None):
     return new_blocks
 
 
+def _tuples_to_blocks_no_consolidate(tuples, dtype: Optional[DtypeObj]) -> List[Block]:
+    # tuples produced within _form_blocks are of the form (placement, whatever, array)
+    if dtype is not None:
+        return [
+            make_block(
+                np.atleast_2d(x[2].astype(dtype, copy=False)), placement=x[0], ndim=2
+            )
+            for x in tuples
+        ]
+    return [make_block(np.atleast_2d(x[2]), placement=x[0], ndim=2) for x in tuples]
+
+
```

Comment:
So regardless of whether you consolidate or not, a block is created for every column. Doesn't that mean that, upon later consolidation, a 2D array would have to be reconsolidated, even though it's already contiguous? Is this any different from how `create_block_manager_from_blocks` creates one single block from a numpy array? 

```python
        if len(blocks) == 1 and not isinstance(blocks[0], Block):
            # if blocks[0] is of length 0, return empty blocks
            if not len(blocks[0]):
                blocks = []
            else:
                # It's OK if a single block is passed as values, its placement
                # is basically "all items", but if there're many, don't bother
                # converting, it's an error anyway.
                blocks = [
                    make_block(
                        values=blocks[0], placement=slice(0, len(axes[0])), ndim=2
                    )
                ]
```

Reply:
> So regardless of whether you consolidate or not, a block is created for every column

I think this is only correct if the arrays that go in to _form_blocks are each single-column to begin with.  Am I wrong on that?

As it is in this PR, we only get `consolidate=False` here in cases where a dict is passed, which should all be single-column.

> Doesn't that mean that, upon later consolidation, a 2D array would have to be reconsolidated, even though it's already contiguous?

I think the only case where we would get already-contiguous data split into separate blocks would be the case you're addressing in #39272 with object dtype.  In that case we would get here with consolidate=True, so I don't think that would be an issue.

Reply:
I see, I don't think this PR will make #39272 any less complicated then.
Note that #39272 also made `test_constructor_series_copy` fail.